### PR TITLE
feat: Introduce `<Autocomplete>` component (alpha)

### PR DESCRIPTION
**Pull request by sebald**

### CODE REVIEW

File: packages/components/src/Autocomplete/Autocomplete.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          <Autocomplete.Item key="lord-of-the-rings">
+            Lord of the Rings
+          </Autocomplete.Item>
+          <Autocomplete.Item key="star-wars">Star Wars</Autocomplete.Item>
+          <Autocomplete.Item key="star-trek">Star Trek</Autocomplete.Item>
+          <Autocomplete.Item key="firefly">Firefly</Autocomplete.Item>
+        </Autocomplete>
+        <pre>current: {current}</pre>
+        <pre>
+          submitted: (key: {submitted[0]}, value: {submitted[1]})
```
Comment (sarahgm) [inline packages/components/src/Autocomplete/Autocomplete.stories.tsx:102]: in my story the value is empty. should this be?

File: packages/components/src/Autocomplete/Autocomplete.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import React, { Key, useState } from 'react';
+import { Meta, ComponentStory } from '@storybook/react';
+
+import { Container } from '../Container';
+import { Stack } from '../Stack';
+import { Autocomplete } from './Autocomplete';
+import { useAsyncList } from '@react-stately/data';
+
+export default {
```
Comment (sarahgm) [inline packages/components/src/Autocomplete/Autocomplete.stories.tsx:9]: error ia not working

File: docs/src/demos/components/Autocomplete/autocomplete-controlled.demo.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    () =>
+      vegetables.filter(item =>
+        item.name
+          .toLocaleLowerCase()
+          .replace(/\s/g, '')
+          .includes(input.toLocaleLowerCase().replace(/\s/g, ''))
+      ),
+    [vegetables, input]
+  );
+  console.log(filteredVeggies);
```
Comment (sarahgm) [inline docs/src/demos/components/Autocomplete/autocomplete-controlled.demo.tsx:32]: remove this?

File: packages/components/src/Autocomplete/Autocomplete.stories.tsx (hunk 2 above)
Comment (sebald) [reply packages/components/src/Autocomplete/Autocomplete.stories.tsx:9]: fixed

File: docs/src/demos/components/Autocomplete/autocomplete-controlled.demo.tsx (hunk 3 above)
Comment (sebald) [reply docs/src/demos/components/Autocomplete/autocomplete-controlled.demo.tsx:32]: removed

**Review by sarahgm — APPROVED** [review]

**Review by sarahgm — APPROVED** [review]

**Pull request merged**